{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyN12/xiZkD2WyeM3yu+57sE",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "ir",
      "display_name": "R"
    },
    "language_info": {
      "name": "R"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/miqrom29/WGSExtract-Dev-Experimental/blob/master/RhaploCart.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 472
        },
        "id": "d28c3c82",
        "outputId": "71c674ab-3cb2-4024-c20b-f5f678385be1"
      },
      "source": [
        "print(\"Hello from R in Colab!\")\n",
        "x <- 1:10\n",
        "print(x)\n",
        "plot(x, main=\"Simple R Plot\")"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[1] \"Hello from R in Colab!\"\n",
            " [1]  1  2  3  4  5  6  7  8  9 10\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Plot with title “Simple R Plot”"
            ],
            "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAAC/VBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHDw8PExMTFxcXGxsbHx8fIyMjJycnKysrL\ny8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd\n3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v\n7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7///+wuh1O\nAAAACXBIWXMAABJ0AAASdAHeZh94AAAeJElEQVR4nO3deXxU9bnH8Sd7wioEFBAIFOziAgG1\nXIpY3AVEW722gloVbLWoVS4gLhXUFm2lLtBaa11b9V6tCLXWrVi1itoCLVRkEZdWLAhKggJC\nQpLzumdmMkmAMBDmO+c3Z+bz/mPOb2ZOznm4dz5M5iRS8wAkzVwPAGQCQgIECAkQICRAgJAA\nAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFC\nAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIE\nCAkQICS3Hjcr0u+awkOgeYQUtIpbjutaWNDxyMn/itzThHSP1WvV59xXmnm8uOeZz7fwbGgZ\nQgrYX0rjL/ri3/t3V91++8/38iv3JqSISc0/PrZut4f4KM+Wt+xPgZ0RUrAq/I6Kx9xw9fBc\ns7brW/SlewjppJEjR570lUgyD+/8+PB+/tns1t0eYpYRUrIIKVj3m+UviSzm+i/tn7XoS/cQ\nUmV09XKJ2eBdH/9nZ7Puuz3EEEJKGiEFa5pZ79jqx7fOfb/hpX2f2VDv8fKSsmurvWWj9mt9\nwlL/wXvNDvYeG9Ku7fHzI1/QUMGKi/oWtT1i5vbGozaG5F1kVtLM4z/xV6sbD/Hp9EEdCvY/\n8f4afz0y9q3f5an8Y2c+QgrWL81yn21yv/6l/ajZob/LibyeL3mvU2TTeaPnPWzWbUb0RZ7/\nfOOu3hPFsVf+sVsbjtIkpKn+3s08/pi/eqvhEIsPrP/gNPgTQtIgpGCtzDPLG3H3W3X19+tf\n2v6ma/dTL23v3xve7fJB/uv69uiLv6jovP+7qY1ZWXXDru/537xNXrng62ZXNxy1SUjDzL7S\nzOM3muVsiB+iwu+o912/n5JvNtLz3nrS3+uRV94L5M+fsQgpYDfF/v7vMOrXn0XuNoZkZ/pv\nNpGLee96VV8yGxV7cKwXfbuy5xt2vdRsmL/5uI21bXhLigdTs+oSfzXL2/lxb0mp2eENh7jB\nrN1//O1D/tMLPW+t8RkpaYQUtGeH5sRa2u8Bb4eQ/uZ51UVmo/27k8zKYw8u9u9tb2t2fcOu\nfcyu3eo72uyF+DF3uPx9dJXX9PERp5126kD/vcfmNhyiv9n5kadrOpjdSEgShBS8D397Ub9o\nTE/sENLn/ra72c3+5lazPtEHC2ojXzDA7IL4rnW5jcnMjB+xSUgdr2z86NT08ZzpXsMh8qPX\nwn2Do90SkgAhubH6mrzINbkmIUWvAvhvN3f5m5/HQ9ovuvNRZt+O77O5yXvPtPjR6t95TvPf\nYb7R9CzxkAp7nP1Xz2t6iLujzx9vdgohSRBS0KrrrzNMj74J7SGk2MXqAdHPSvVvJ36Ad+x8\nzPhnofv87exmHo9rfEeaEb0/yOxsQpIgpEC9e/qX8+fFlvf7L9+tewrJVvn3treJfpSp3+eL\nzVypjgdT9zWzrht3fTyu/hDl0X78pv0PXz8lJAlCCtTWjmZlb0ZW6wZEr1PvKaSrvNi1vBcb\n9rnMrNsWz6sZc8FVH8YP2xDMYv/96ruNp9tNSDeatV7tb3/tf3Ra6Xkf+Xs1/VVX7ANCCtYv\nI5/7B59z9rBCf3HvHkPKL5j20i9Lzb5U07DPuyVmQ55+/nSzQ2riR20M5nL/8C97uz4eU3+I\nyu5mfe98YnKR2YX+3ZoCs6G/ez71f/hMRkgBuyk/fqkgb6q3x5A6XBXdtXh+k30eL4p9/YGN\n3441BvNpV7Mv7vLzpbhdfrPh9OiuwyPLkan8U2c+Qgrau9d+rXNBYechV62I3NtDSK29u8uL\nO5y6pMmunrdsbO+iVodeW9F4zCbBPGJNfuVhdyF5n/74yPYFXb/5h9jDH35jv+Le01Pyp80a\nhJS++M/wQoSQ0hchhQghpS9CChFCSl+EFCKElL4IKUQICRAgJECAkAABQgIECAkQICRAgJAA\nAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFC\nAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQCCAkBYvBEJlcctf5akPaYEBIbOg\nxS/z1Ic036pSfg5AqMrmt/hrCAnYCSEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQECQYdU\n9+68OXP+/MEe9iIkhEywIVVM3D/2C349b/w80X6EhJAJNKQ1ve2g86fdcssPR3ez/hUJdiQk\nhEygIY0r+F39qubOnMsT7EhICJlAQ+oytnH97R4JdiQkhEygIRVMb1xfX5hgR0JCyAQaUtm3\nGten9UqwIyEhHS2ZMmLElCXNPhVoSJfnzNgWW22ealMS7EhISEM35x09efLReTc391ygIVUO\ntLbHnX/pJecNa2VDNyXYkZCQfh4vnBPZzCmc3cyTwf4cqeq28rzIj5EK/uvXNYn2IySknwGT\nYttJA5t5MvBfEdr69qJFq5rLpPaleQ3uICSkm032emzxWs7mXZ919rt2n6za6YH3Ondo0Mqa\nGRVwaY2tjC1W2Jpdn3UW0pRER/mVJfoEBThQXfx0bPHH4updnyUkYO+ccWJdZFN7whnNPElI\nwN5Z0X7MWs9bO6b9ymaeDDSkw5voQkgImUWHWq9eduii5p4LNKTc3KIGeYSEsKld9OCDC2ub\nfSrQkKa0bbxUx7d2yCSBhlQ94IiG6x2EhEwS7MWGZSWT4ktCQiYJ+Krdpxviq5eb/dW/eoSE\nkEnPf0WIkBAyhAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIg\nQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBI\ngAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAA\nISErbN2e2uMTEjLf5qsPyisacFddCk9BSMh4Ff16z3r1hevbfbs2decgJGS8cYdURjZL296b\nunMQEjLd5pInY4urv5q6kxASMt1iq4gtnilJ3UkICZnu77YxtniuOHUnISRkus8Kn4strh+Q\nupMQEjLemCO2RDbvdZyVunMQEjLe2j6HPbx80az9T65O3TkICZlvw8WlZmXTU9gRISE7rNuY\n2uMTEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQI\nEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBAS\nIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBA\nSIAAIQEChAQIOAmpeunCbQl3ICTE1Lw1+4WPXQ+xN4IN6c/Deg1/w3uum1m7OxPtR0iI+lMf\n61SY951K13PsWaAhvZZv7XJbv9aux3e+1cGeTbAjISHi+YIJ//G2v/SVQVWuJ9mjQEMa1WWJ\nt/6Ynv0/97yKXicn2JGQ4Kvre0V0u67zLxxPsmeBhlT6I/9mgT0YWf+4Y4IdCQm+RTlrY4sr\nh7kdZC8EGlL+b/2bNfZ0ZH1f/k5P/vuLX2jQiZDgeXPif9k+0NvpHHsj0JAOmObfvGwzI+tr\nDtjpyeqH7m5wNiHB/4hUVBNbzDzU7SB7IdCQzur4YtU/D/tKzw89b1mH/06wI9/awbch/5nY\n4vgL3Q6yFwINaXlbM+u4rKzVMYPz8/6aYEdCQsRFX3g/srm9YKnjQfYs2J8jvTl60PkrvDe/\nmmNf+H2i/QgJEVtOaDv29qlHFT/iepA9c/MrQpvWJ36ekBBV+8jo/sdOWOl6jL3A79oBAoQE\nCBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQ\nEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIg\nQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBI\ngAAhAQKEhJR4Y/q5Vz5W5XqK4BASUqDqnNzBFwxv/+UVrgcJDCEhBcZ3W+jfVp7Sa7PrSYJC\nSNBbnTcvut1y4B2OJwkMIUHvoS51scWlp7odJDiEBL07+tcvfjzE6RwBIiToPVpaG1t87wy3\ngwSHkKC3vujx6HZD6b2OJwkMISEFprb/g3/73uB+WfOTJEJCCtRdld/jhH55X1/jepDAEBJS\n4l8PXjur5S+t8CIkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQ\nAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAAB\nQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUIC\nBAgJECAkQICQAAFCAgQICRAgJEDARUg1/5z/QeI9CMmZNROO6FB+0TuuxwidYEOaf4l/89AB\nZtb/L4n2IyRX/tGp/Ja5dxzV5gXXg4RNoCG9VNimznvc2pw5/oTcooUJdiQkR6r6jtnub+om\nlla4HiVkAg1p2P6rPK932Rp/+UbJqAQ7EpIjc1tvjG6re/zC8SRhE2hI7SZ53kabGV1/d7+d\nntxy608afJOQ3LjumPrFORc4nSN8Ag2p9XWety3niej6huKdnlxz1OENetpn+3oOJOPK4fWL\nC892Okf4BBrSkIO2eN7XJkWW2/r3T7Aj39o5cs+BtbHFkdPcDhI6gYb0lA18fvuirr/ZUv3G\nsXZ3gh0JyZH1be6Mbp/MW+Z4krAJ9vL3Pa2t5OAyy8uznP+pS7AfIblyT/5Vy6veubnketeD\nhE3AP5D9aMZJZW2LSg//waKEuxGSM3MPMrPu97keI3T4FSHsZO2re/i9EzSDkAABQgIECAkQ\nICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAk\nQICQAIFkQ3olvqi9TTFOPUJCyCQbUu6Ez6Pbt4co36cICSGTbEjD7Uuv+29Hd5QU3agbipAQ\nNkl/RprdI3fym0PtmLdlI3mEhNBJ/mLD5kn5VvqAap4YQkLIJB/StmlFdsCjqnliCAkhk3RI\n8w7KGb9osI38t2wkj5AQOsmGNNp6veh5tTOKW98um4mQEDrJhpRzcewlv2Iwl7+RxZIN6YX4\novZninHqERJChl8RAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQI\nCRAgJECAkAABQgIECAkQICRAgJAAAUICBAgpc3zw6lrXI2QvQsoU93Y3s4Pmuh4jWxFShphW\ncvM7Vcun5N/jepAsRUiZYVnek9HtnW3WOZ4kSxFSZph2ZGxb2423JCcIKTOc/d36xfApTufI\nWoSUGS44t34x7Dqnc2QtQsoMv+hRHd1ubP17x5NkKULKDBWlE+v8zfYxfatcj5KdCClDzGsz\n9I65t5R3+ofrQbIUIWWKdy4q73DEhDWux8hWhAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKE\nBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQI\nEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBAS\nIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECLgJ6dMpyxM+n10hzZ917YP/\ncj0EkuMmpNX2VMLnsymk/3w9r98JPfKvrnM9CJIRaEjj4kbbiePGJdgxi0Kq6jf4PX/zh/ZT\nXU+CZAQaku0gwY5ZFNI9nTZEt48XrXc8CZIRaEgT8sqfq4x4yx6trNz52c8qGtyWPSGdflFs\nW1v6qNtBkJRgPyMtKM/5/kav+c9I7+Q0fbv6bJ/PETJDptcv+s90OgeSE/DFhu0/Kek2ezcX\nG5YubHBN9rwjjbostq074GG3gyApgV+1e+c4G/UBV+0azDxwS3Q7L+9Dx5MgGQ4ufz/Qsc00\nQorbXDYq8mlxYbdLXE+CZLj4OdK6s4yQGqz4cvvhFwzOPbfK9SBIhpsfyD4zcVnC57MpJK/q\nsSvPnf6G6ymQHH7XDhAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAg\nJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRA\ngJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQ\nAAFCAgQICRAgJECAkAABQgIECAkQIKR9t3LCsf3PeqTW9RhIB4S0zx4pPmrq7WPbnLDF9SBI\nA4S0r5YW3B7ZvN/7IteTIA0Q0r668PjY9pn8DW4HQTogpH11yMzYdnvR824HQTogpH3V64H6\nRYc5TudAWiCkfTVsSmy7NufvbgdBOiCkffWLzuui2yv61jmeBGmAkPZV1VcPfnm7958rCv7k\nehKkAULaZ5Xn5hV2sr50BI+QkvLxC7PfqnE9BNICIQEChAQIEBIgQEiAACEBAoQECBASIEBI\ngAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAA\nIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEB\nAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAgLOQKt5P8GTKQ9q4LrXHR7YJ\nNqQlI8qOurMmupyS6CipDal6eplZ6cUbUngKZJtAQ3q1yFoV2NcrImt3IVWfvP+sRcsfPqzP\nR6k7B7JNoCGNLJhbt+22giM3ey5DmlX6XmSz5fAxqTsHsk2gIfU4J3L758IRNS5DKr8+tn2u\n8LPUnQRZJtCQCqZGN7+1HzQT0idnn9ngcEvha7z4udh2o/0jdSdBlgk0pO6nxrZX2y27hrTx\nsu81GGFV+3qOPSt5JratsMWpOwmyTKAh/SDn59WRbd15dsVliY4yP5UhffWa2PbJks2pOwmy\nTKAhfdLTjo8u6n5g5iyke9otjWwqD74wdedAtgn250gfj7+ifvVEH2ch1X6r3fUvvDqrd7+K\n1J0D2SY9f0UopSF5dXcNLMo76Bq+sYNONobk2741tcdHtsnSkAAtQgIECAkQICRAgJAAAUIC\nBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQI\nCRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQ\nICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAIFwh1S588MGFtSk/\nPdBSoQpp0aHWq5cduijl5wdaKEwhrWg/Zq3nrR3TfmXKBwBaJkwhnX5iXWRTe+IZKR8AaJkQ\nhVRd/MfY4qni6pRPALRIiEJaY/Xf0q2wNSmfAGiREIW0yV6PLV7L2ZzyCYAWCVFI3sDJse2k\ngSkfAGiZMIX0eOGcyGZO4eyUDwC0TJhC8m7OO3ry5KPzbk75+YEWClVI3pIpI0ZMWZLy0wMt\nFa6QgDRFSIAAIQEChAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQI\nEBIgkJ4hLTAgZBa0+GWe+pC8xQtT67xDHkojE4tcT9DUr+wm1yM01XWs6wmaOvrk3b2kFrf8\nVR5ASKn2w+NdT9DUH1u7nqCpDZZW/yrGl37leoKmzj9feDBCEiOkBAgpnRHS7hFSAoS0I0La\nPUJKgJB2REi7R0gJENKOCGn3CCkBQtoRIe0eISVASDsipN0jpAQIaUeEtHuElAAh7YiQdo+Q\nEiCkHd0wwvUETf2pg+sJmvosZ7nrEZo67H7XEzT1ve8JD5YBIW3+yPUETdW+73qCHbzreoAd\nrE6r/xCgokJ4sAwICXCPkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECA\nkAABQgIECAkQCH1IFRN7FvY67XXXYzQxwca5HiHumaPbtD/mJddT1Ft+Tpf8Tt/4q+sxfNVX\n5R4eW1VeXlbQddwawTHDHtKGXjbyurPzi//pepAGC/LSJqT7rc8PJ3UubPn/2E8qLG3bcepv\nf9Ql/8+uB/GWDWxbH1LVQDtj+tiC3oL/VDbsIV1iP/dvn7C0+Xcbtpf3T5eQ1rUZsNnzVrUZ\n73qQqDH2on+7xIa5HuTTkiNWFcVCus1+6t8+ZhOTP2rYQ7riuGr/tq6kzPUgcT/JeTZdQpph\nz0U2da7niBlkkf9Pee16uR5kw8Rqrz6k8rbbIpu++yf/f6OwhxSzrWCI6xHqvVPy/cp0Cemk\nkmpv26eup4g7z970bz/OHe56kIhYSFvzjoveO9+S/zdiMiOkmdFv8NLBcV03pk1IZQf/fUiO\n9XnA9Rwxyzr0f2Xt349r9YbrQSJiIb1tsX/ZbprNS/qIGRHSy4VHbXc9Q8wDNttLm5DalnWd\nOHtmT3vE9SAxKw42s56vuR4jKhbSIrskem+GzUn6iJkQ0v8WDdzgeoaYdR1P8dInpCL7jX+7\npk2XGteTRCzr3ePWp+47pH3yf/kLxEO6NHrvFpub9BHDH1LdVDv5M9dD1Durzb/TKKTSvC2R\nzZmWFj8b+K9WH/q3Ww48sNr1JF48pFV2XvTeD+2FpI8Y+pDqxtplafE3ru8Zu2716tVv2ejV\nafER//C86Gt2vKXDD5I25RwT3X7HljqeJCIWUlV+7Fr8aPt30kcMfUiX202uR2gw0eKmuB4l\n4lKLfrA/0T5wPYlvvQ2Obr9lCx1PElF/+XtQq8h7dm23HskfMewhPWGXux6h0bKnIh61E59K\ni3+6fmHOsds8b0FuP9eDRPUuWOnfVnZst831JF5DSL+26/3bu+yG5I8Y9pD62GVTopT/IHpy\n0uYzkneFld/w3ZLC9Phluzm5pdfeP7233el6kJf9V0teF//mE69mqJ12w1k5h21J/qhhD6nh\nm6n3XU/SIH1CqvtV/+L2I/7meox6r32jc36H4592PYZ3c/wls8r/5DaprODASxTXfMMeEpAW\nCAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJ\nECAkQICQAAFCAgQICRAgJECAkAABQgIECAkQICRAgJAAAUICBAgJECAkQICQAAFCAgQIKczy\nBrmeAPUIKQyW20nNPk5IaYOQwoCQ0h4hhQEhpT1CCoNYSKNt05Vlhd1vq/PXTw8s7jyuMhrS\nR+N7FnQ67W+eNy9ndGTn4bmvOB02OxFSGMRCOs9Ouvj1+Sfa/Z73al63m+45Z2iBH9L6svZT\nHrqpe9HLnnexzfO82TbB9bjZiJDCIBbSOIu847xrp/jvOua/A3njzQ/p+/kL/OUHbY/wvE29\nDtq2uccXP3c7bHYipDCIh/Rc5E6rcq+2pE9k9Q8/pLpOA9dGnGSbPO/FnGmTcl9zOmq2IqQw\niIe0LHKn/SHeh3ZCZLXVD+kji3vLf2R8UcFkl4NmL0IKg3hIqyJ3/JDetlHRx3MGeaus/NmY\nSv+BRWZvOpwzixFSGOwc0urYO9Km6DtSeeN+tYMPKB1a52TEbEdIYbBzSNsL+0ZW8yMXGzoV\nR96KvPWRmxn26AN2h7MxsxkhhcHOIXnDolftxkSv2tk1/nJ9l1M8b2XJCM87ptXbTmfNUoQU\nBruE9EzO/lfNOOXY9n5I63raBQ/e1LPgT/43dq3/5ddUNKTW8bjZiJDCYJeQvEcPK+w8trLH\nAH+59vs98vc79a+e9zO7LfL8jXarw1GzFSEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEC\nhAQIEBIgQEiAACEBAoQECBASIEBIgAAhAQKEBAgQEiBASIAAIQEChAQIEBIgQEiAACEBAoQE\nCBASIEBIgAAhAQKEBAgQEiDw/0GYpuNYo8G3AAAAAElFTkSuQmCC"
          },
          "metadata": {
            "image/png": {
              "width": 420,
              "height": 420
            }
          }
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# En una celda de Colab con prefijo !\n",
        "system('wget https://github.com/grenaud/vgan/releases/download/v3.1.0/ -O vgan')\n",
        "system('chmod +x vgan')\n",
        "system('mkdir -p $HOME/bin')\n",
        "system('mv vgan $HOME/bin/')\n",
        "system('mkdir -p $HOME/share/vgan/hcfiles')\n",
        "# ... y descargar los hcfiles desde el wiki/manual de vgan"
      ],
      "metadata": {
        "id": "Z4lISmzNSnxF"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "system(\"echo $PATH\")  # Debe mostrar /home/.../bin\n",
        "system(\"which vgan\")  # Debe mostrar ~/bin/vgan\n",
        "system(\"vgan --help | head -20\")  # Muestra uso general de vgan\n"
      ],
      "metadata": {
        "id": "U4hIOgLNU2c9"
      },
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "system(\"vgan haplocart --help | head -30\")\n",
        "# Si sale el help detallado de haplocart (opciones como -i input.fastq.gz, --hcfiles, etc.), ¡está perfecto!\n"
      ],
      "metadata": {
        "id": "ztkA4PyKU_OO"
      },
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Verificar PATH y ubicación del binario\n",
        "system(\"echo $PATH\", intern = TRUE)\n",
        "system(\"which vgan\", intern = TRUE)\n",
        "system(\"ls -la ~/bin/\", intern = TRUE)\n",
        "\n",
        "# 2. Test directo del ejecutable\n",
        "system(\"file ~/bin/vgan\", intern = TRUE)\n",
        "\n",
        "# 3. Verificar directorios creados\n",
        "system(\"ls -la ~/share/vgan/\", intern = TRUE)\n",
        "system(\"ls -la ~/share/vgan/hcfiles/\", intern = TRUE)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 263
        },
        "id": "kL7w9umuZNm2",
        "outputId": "355cca91-bf1f-4bce-8820-9673168b4ff1"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'/opt/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin'"
            ],
            "text/markdown": "'/opt/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin'",
            "text/latex": "'/opt/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin'",
            "text/plain": [
              "[1] \"/opt/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/tools/node/bin:/tools/google-cloud-sdk/bin\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Warning message in system(\"which vgan\", intern = TRUE):\n",
            "“running command 'which vgan' had status 1”\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)\n",
              "attr(,\"status\")\n",
              "[1] 1\n",
              "attr(,\"errmsg\")\n",
              "[1] \"Resource temporarily unavailable\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'total 12'</li><li>'drwxr-xr-x 2 root root 4096 Feb 14 11:44 .'</li><li>'drwx------ 1 root root 4096 Feb 14 11:44 ..'</li><li><span style=white-space:pre-wrap>'-rwxr-xr-x 1 root root    0 Feb 14 11:44 vgan'</span></li></ol>\n"
            ],
            "text/markdown": "1. 'total 12'\n2. 'drwxr-xr-x 2 root root 4096 Feb 14 11:44 .'\n3. 'drwx------ 1 root root 4096 Feb 14 11:44 ..'\n4. <span style=white-space:pre-wrap>'-rwxr-xr-x 1 root root    0 Feb 14 11:44 vgan'</span>\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item 'total 12'\n\\item 'drwxr-xr-x 2 root root 4096 Feb 14 11:44 .'\n\\item 'drwx------ 1 root root 4096 Feb 14 11:44 ..'\n\\item '-rwxr-xr-x 1 root root    0 Feb 14 11:44 vgan'\n\\end{enumerate*}\n",
            "text/plain": [
              "[1] \"total 12\"                                     \n",
              "[2] \"drwxr-xr-x 2 root root 4096 Feb 14 11:44 .\"   \n",
              "[3] \"drwx------ 1 root root 4096 Feb 14 11:44 ..\"  \n",
              "[4] \"-rwxr-xr-x 1 root root    0 Feb 14 11:44 vgan\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'/root/bin/vgan: empty'"
            ],
            "text/markdown": "'/root/bin/vgan: empty'",
            "text/latex": "'/root/bin/vgan: empty'",
            "text/plain": [
              "[1] \"/root/bin/vgan: empty\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'total 12'</li><li>'drwxr-xr-x 3 root root 4096 Feb 14 11:44 .'</li><li>'drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..'</li><li>'drwxr-xr-x 2 root root 4096 Feb 14 11:44 hcfiles'</li></ol>\n"
            ],
            "text/markdown": "1. 'total 12'\n2. 'drwxr-xr-x 3 root root 4096 Feb 14 11:44 .'\n3. 'drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..'\n4. 'drwxr-xr-x 2 root root 4096 Feb 14 11:44 hcfiles'\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item 'total 12'\n\\item 'drwxr-xr-x 3 root root 4096 Feb 14 11:44 .'\n\\item 'drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..'\n\\item 'drwxr-xr-x 2 root root 4096 Feb 14 11:44 hcfiles'\n\\end{enumerate*}\n",
            "text/plain": [
              "[1] \"total 12\"                                        \n",
              "[2] \"drwxr-xr-x 3 root root 4096 Feb 14 11:44 .\"      \n",
              "[3] \"drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..\"     \n",
              "[4] \"drwxr-xr-x 2 root root 4096 Feb 14 11:44 hcfiles\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'total 8'</li><li>'drwxr-xr-x 2 root root 4096 Feb 14 11:44 .'</li><li>'drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..'</li></ol>\n"
            ],
            "text/markdown": "1. 'total 8'\n2. 'drwxr-xr-x 2 root root 4096 Feb 14 11:44 .'\n3. 'drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..'\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item 'total 8'\n\\item 'drwxr-xr-x 2 root root 4096 Feb 14 11:44 .'\n\\item 'drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..'\n\\end{enumerate*}\n",
            "text/plain": [
              "[1] \"total 8\"                                    \n",
              "[2] \"drwxr-xr-x 2 root root 4096 Feb 14 11:44 .\" \n",
              "[3] \"drwxr-xr-x 3 root root 4096 Feb 14 11:44 ..\""
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Add ~/bin to the R session's PATH so vgan can be found\n",
        "Sys.setenv(PATH = paste0(Sys.getenv(\"HOME\"), \"/bin:\", Sys.getenv(\"PATH\")))\n",
        "\n",
        "# Test 1: vgan general\n",
        "system(\"vgan --version\", intern = TRUE)\n",
        "\n",
        "# Test 2: HaploCart específico (¡lo más importante!)\n",
        "system(\"vgan haplocart --help\", intern = TRUE)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "id": "GA_tfMD2ZYGm",
        "outputId": "24589fac-485c-4a70-8a36-88540d15c538"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. AÑADIR ~/bin al PATH permanentemente para esta sesión R\n",
        "Sys.setenv(PATH = paste0(Sys.getenv(\"PATH\"), \":/root/bin\"))\n",
        "\n",
        "# 2. LIMPIAR archivos corruptos/vacíos\n",
        "system(\"rm -f ~/bin/vgan /content/vgan\", intern = TRUE)\n",
        "\n",
        "# 3. RE-DESCARGAR vgan CORRECTAMENTE (usa Bioconda, más fiable que GitHub)\n",
        "system(\"mkdir -p ~/bin\", intern = TRUE)\n",
        "system(\"curl -Ls https://micro.mamba.pm/api/micromamba/linux-64/latest | tar -xvj bin/\", intern = TRUE)\n",
        "system(\"./bin/micromamba create -n vgan -c bioconda vgan --root-prefix=/tmp/vgan-env -y\", intern = TRUE)\n",
        "system(\"find /tmp/vgan-env -name vgan -type f -executable -exec cp {} ~/bin/vgan \\\\;\", intern = TRUE)\n",
        "\n",
        "# 4. VERIFICAR que ahora SÍ funciona\n",
        "system(\"ls -lh ~/bin/vgan\", intern = TRUE)\n",
        "system(\"file ~/bin/vgan\", intern = TRUE)\n",
        "system(\"vgan --version\", intern = TRUE)\n",
        "system(\"vgan haplocart --help | head -15\", intern = TRUE)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 402
        },
        "id": "v_StjapHZ3HA",
        "outputId": "cdbd83fe-fe35-4de3-9a40-aa099530b088"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'bin/micromamba'"
            ],
            "text/markdown": "'bin/micromamba'",
            "text/latex": "'bin/micromamba'",
            "text/plain": [
              "[1] \"bin/micromamba\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>''</li><li>''</li><li>'Transaction'</li><li>''</li><li><span style=white-space:pre-wrap>'  Prefix: /tmp/vgan-env/envs/vgan'</span></li><li>''</li><li><span style=white-space:pre-wrap>'  Updating specs:'</span></li><li>''</li><li><span style=white-space:pre-wrap>'   - vgan'</span></li><li>''</li><li>''</li><li><span style=white-space:pre-wrap>'  Package  Version  Build       Channel      Size'</span></li><li>'───────────────────────────────────────────────────'</li><li><span style=white-space:pre-wrap>'  Install:'</span></li><li>'───────────────────────────────────────────────────'</li><li>''</li><li><span style=white-space:pre-wrap>'  + vgan     1.0.2  h9ee0642_0  bioconda      7MB'</span></li><li>''</li><li><span style=white-space:pre-wrap>'  Summary:'</span></li><li>''</li><li><span style=white-space:pre-wrap>'  Install: 1 packages'</span></li><li>''</li><li><span style=white-space:pre-wrap>'  Total download: 7MB'</span></li><li>''</li><li>'───────────────────────────────────────────────────'</li><li>''</li><li>''</li><li>''</li><li>'Transaction starting'</li><li>'Linking vgan-1.0.2-h9ee0642_0'</li><li>''</li><li>'Transaction finished'</li><li>''</li><li>''</li><li>'To activate this environment, use:'</li><li>''</li><li><span style=white-space:pre-wrap>'    micromamba activate vgan'</span></li><li>''</li><li>'Or to execute a single command in this environment, use:'</li><li>''</li><li><span style=white-space:pre-wrap>'    micromamba run -n vgan mycommand'</span></li><li>''</li></ol>\n"
            ],
            "text/markdown": "1. ''\n2. ''\n3. 'Transaction'\n4. ''\n5. <span style=white-space:pre-wrap>'  Prefix: /tmp/vgan-env/envs/vgan'</span>\n6. ''\n7. <span style=white-space:pre-wrap>'  Updating specs:'</span>\n8. ''\n9. <span style=white-space:pre-wrap>'   - vgan'</span>\n10. ''\n11. ''\n12. <span style=white-space:pre-wrap>'  Package  Version  Build       Channel      Size'</span>\n13. '───────────────────────────────────────────────────'\n14. <span style=white-space:pre-wrap>'  Install:'</span>\n15. '───────────────────────────────────────────────────'\n16. ''\n17. <span style=white-space:pre-wrap>'  + vgan     1.0.2  h9ee0642_0  bioconda      7MB'</span>\n18. ''\n19. <span style=white-space:pre-wrap>'  Summary:'</span>\n20. ''\n21. <span style=white-space:pre-wrap>'  Install: 1 packages'</span>\n22. ''\n23. <span style=white-space:pre-wrap>'  Total download: 7MB'</span>\n24. ''\n25. '───────────────────────────────────────────────────'\n26. ''\n27. ''\n28. ''\n29. 'Transaction starting'\n30. 'Linking vgan-1.0.2-h9ee0642_0'\n31. ''\n32. 'Transaction finished'\n33. ''\n34. ''\n35. 'To activate this environment, use:'\n36. ''\n37. <span style=white-space:pre-wrap>'    micromamba activate vgan'</span>\n38. ''\n39. 'Or to execute a single command in this environment, use:'\n40. ''\n41. <span style=white-space:pre-wrap>'    micromamba run -n vgan mycommand'</span>\n42. ''\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item ''\n\\item ''\n\\item 'Transaction'\n\\item ''\n\\item '  Prefix: /tmp/vgan-env/envs/vgan'\n\\item ''\n\\item '  Updating specs:'\n\\item ''\n\\item '   - vgan'\n\\item ''\n\\item ''\n\\item '  Package  Version  Build       Channel      Size'\n\\item '───────────────────────────────────────────────────'\n\\item '  Install:'\n\\item '───────────────────────────────────────────────────'\n\\item ''\n\\item '  + vgan     1.0.2  h9ee0642\\_0  bioconda      7MB'\n\\item ''\n\\item '  Summary:'\n\\item ''\n\\item '  Install: 1 packages'\n\\item ''\n\\item '  Total download: 7MB'\n\\item ''\n\\item '───────────────────────────────────────────────────'\n\\item ''\n\\item ''\n\\item ''\n\\item 'Transaction starting'\n\\item 'Linking vgan-1.0.2-h9ee0642\\_0'\n\\item ''\n\\item 'Transaction finished'\n\\item ''\n\\item ''\n\\item 'To activate this environment, use:'\n\\item ''\n\\item '    micromamba activate vgan'\n\\item ''\n\\item 'Or to execute a single command in this environment, use:'\n\\item ''\n\\item '    micromamba run -n vgan mycommand'\n\\item ''\n\\end{enumerate*}\n",
            "text/plain": [
              " [1] \"\"                                                        \n",
              " [2] \"\"                                                        \n",
              " [3] \"Transaction\"                                             \n",
              " [4] \"\"                                                        \n",
              " [5] \"  Prefix: /tmp/vgan-env/envs/vgan\"                       \n",
              " [6] \"\"                                                        \n",
              " [7] \"  Updating specs:\"                                       \n",
              " [8] \"\"                                                        \n",
              " [9] \"   - vgan\"                                               \n",
              "[10] \"\"                                                        \n",
              "[11] \"\"                                                        \n",
              "[12] \"  Package  Version  Build       Channel      Size\"       \n",
              "[13] \"───────────────────────────────────────────────────\"     \n",
              "[14] \"  Install:\"                                              \n",
              "[15] \"───────────────────────────────────────────────────\"     \n",
              "[16] \"\"                                                        \n",
              "[17] \"  + vgan     1.0.2  h9ee0642_0  bioconda      7MB\"       \n",
              "[18] \"\"                                                        \n",
              "[19] \"  Summary:\"                                              \n",
              "[20] \"\"                                                        \n",
              "[21] \"  Install: 1 packages\"                                   \n",
              "[22] \"\"                                                        \n",
              "[23] \"  Total download: 7MB\"                                   \n",
              "[24] \"\"                                                        \n",
              "[25] \"───────────────────────────────────────────────────\"     \n",
              "[26] \"\"                                                        \n",
              "[27] \"\"                                                        \n",
              "[28] \"\"                                                        \n",
              "[29] \"Transaction starting\"                                    \n",
              "[30] \"Linking vgan-1.0.2-h9ee0642_0\"                           \n",
              "[31] \"\"                                                        \n",
              "[32] \"Transaction finished\"                                    \n",
              "[33] \"\"                                                        \n",
              "[34] \"\"                                                        \n",
              "[35] \"To activate this environment, use:\"                      \n",
              "[36] \"\"                                                        \n",
              "[37] \"    micromamba activate vgan\"                            \n",
              "[38] \"\"                                                        \n",
              "[39] \"Or to execute a single command in this environment, use:\"\n",
              "[40] \"\"                                                        \n",
              "[41] \"    micromamba run -n vgan mycommand\"                    \n",
              "[42] \"\"                                                        "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'-rwxr-xr-x 1 root root 20M Feb 14 11:45 /root/bin/vgan'"
            ],
            "text/markdown": "'-rwxr-xr-x 1 root root 20M Feb 14 11:45 /root/bin/vgan'",
            "text/latex": "'-rwxr-xr-x 1 root root 20M Feb 14 11:45 /root/bin/vgan'",
            "text/plain": [
              "[1] \"-rwxr-xr-x 1 root root 20M Feb 14 11:45 /root/bin/vgan\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'/root/bin/vgan: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[sha1]=742352d01b7f7edb7c864a71e022e62467527946, for GNU/Linux 3.2.0, stripped'"
            ],
            "text/markdown": "'/root/bin/vgan: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[sha1]=742352d01b7f7edb7c864a71e022e62467527946, for GNU/Linux 3.2.0, stripped'",
            "text/latex": "'/root/bin/vgan: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID{[}sha1{]}=742352d01b7f7edb7c864a71e022e62467527946, for GNU/Linux 3.2.0, stripped'",
            "text/plain": [
              "[1] \"/root/bin/vgan: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[sha1]=742352d01b7f7edb7c864a71e022e62467527946, for GNU/Linux 3.2.0, stripped\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Warning message in system(\"vgan --version\", intern = TRUE):\n",
            "“running command 'vgan --version' had status 1”\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)\n",
              "attr(,\"status\")\n",
              "[1] 1\n",
              "attr(,\"errmsg\")\n",
              "[1] \"Resource temporarily unavailable\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. FIJAR PATH COMPLETAMENTE (incluye TODAS las rutas necesarias)\n",
        "Sys.setenv(PATH = paste0(\"/content/bin:\", Sys.getenv(\"PATH\"), \":/root/bin\"))\n",
        "\n",
        "# 2. VERIFICAR micromamba funciona\n",
        "system(\"which micromamba\", intern = TRUE)\n",
        "system(\"micromamba run -n vgan echo 'Entorno OK'\", intern = TRUE)\n",
        "\n",
        "# 3. EJECUTAR vgan correctamente (¡SIN rutas raras!)\n",
        "system(\"micromamba run -n vgan vgan haplocart --help\", intern = FALSE)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 89
        },
        "id": "g1Nm4_vIbBjT",
        "outputId": "71387865-247d-4959-f48d-ec8f28df1c7a"
      },
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'/content/bin/micromamba'"
            ],
            "text/markdown": "'/content/bin/micromamba'",
            "text/latex": "'/content/bin/micromamba'",
            "text/plain": [
              "[1] \"/content/bin/micromamba\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Warning message in system(\"micromamba run -n vgan echo 'Entorno OK'\", intern = TRUE):\n",
            "“running command 'micromamba run -n vgan echo 'Entorno OK'' had status 1”\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)\n",
              "attr(,\"status\")\n",
              "[1] 1\n",
              "attr(,\"errmsg\")\n",
              "[1] \"Resource temporarily unavailable\""
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. PATH correcto (sintaxis R)\n",
        "Sys.setenv(PATH = paste0(\"/content/bin:/root/bin:\", Sys.getenv(\"PATH\")))\n",
        "\n",
        "# 2. Test directo del ejecutable vgan (SIN micromamba)\n",
        "system(\"ls -la /tmp/vgan-env/envs/vgan/bin/\", intern = TRUE)\n",
        "system(\"/tmp/vgan-env/envs/vgan/bin/vgan --help | head -15\", intern = FALSE)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 69
        },
        "id": "BoD_1gC5boWO",
        "outputId": "45cfcd91-3a3a-496d-f82a-72c951b5e922"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'total 19692'</li><li><span style=white-space:pre-wrap>'drwxr-xr-x 2 root root     4096 Feb 14 11:45 .'</span></li><li><span style=white-space:pre-wrap>'drwxr-xr-x 4 root root     4096 Feb 14 11:45 ..'</span></li><li><span style=white-space:pre-wrap>'-rwxrwxr-x 2 root root 20153528 Jan 23  2023 vgan'</span></li></ol>\n"
            ],
            "text/markdown": "1. 'total 19692'\n2. <span style=white-space:pre-wrap>'drwxr-xr-x 2 root root     4096 Feb 14 11:45 .'</span>\n3. <span style=white-space:pre-wrap>'drwxr-xr-x 4 root root     4096 Feb 14 11:45 ..'</span>\n4. <span style=white-space:pre-wrap>'-rwxrwxr-x 2 root root 20153528 Jan 23  2023 vgan'</span>\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item 'total 19692'\n\\item 'drwxr-xr-x 2 root root     4096 Feb 14 11:45 .'\n\\item 'drwxr-xr-x 4 root root     4096 Feb 14 11:45 ..'\n\\item '-rwxrwxr-x 2 root root 20153528 Jan 23  2023 vgan'\n\\end{enumerate*}\n",
            "text/plain": [
              "[1] \"total 19692\"                                      \n",
              "[2] \"drwxr-xr-x 2 root root     4096 Feb 14 11:45 .\"   \n",
              "[3] \"drwxr-xr-x 4 root root     4096 Feb 14 11:45 ..\"  \n",
              "[4] \"-rwxrwxr-x 2 root root 20153528 Jan 23  2023 vgan\""
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "system(\"/tmp/vgan-env/envs/vgan/bin/vgan haplocart --help\", intern = FALSE)\n"
      ],
      "metadata": {
        "id": "CZGy044LbwgX"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Test completo de vgan\n",
        "system(\"/tmp/vgan-env/envs/vgan/bin/vgan --help\", intern = FALSE)\n",
        "\n",
        "# 2. Test ESPECÍFICO de HaploCart\n",
        "system(\"/tmp/vgan-env/envs/vgan/bin/vgan haplocart --help\", intern = FALSE)\n"
      ],
      "metadata": {
        "id": "Xy1seUt7cA-v"
      },
      "execution_count": 11,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Estructura final para análisis reales\n",
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "system(paste0(ruta_vgan, \" haplocart --help\"), intern = FALSE)\n"
      ],
      "metadata": {
        "id": "RSmEMj5VcRX5"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Instalar samtools en R Colab (vía apt, más simple)\n",
        "system(\"apt update -qq && apt install -y samtools\")\n",
        "\n",
        "# Verificar instalación\n",
        "system(\"samtools --version\", intern = TRUE)\n",
        "\n",
        "\n",
        "# Verificar índice creado\n",
        "system(\"ls -lh /content/mito.*\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 399
        },
        "id": "EkT8A86neknt",
        "outputId": "dba51082-b354-4704-b71f-5d075361f51c"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'samtools 1.13'</li><li>'Using htslib 1.13+ds'</li><li>'Copyright (C) 2021 Genome Research Ltd.'</li><li>''</li><li>'Samtools compilation details:'</li><li><span style=white-space:pre-wrap>'    Features:       build=configure curses=yes '</span></li><li><span style=white-space:pre-wrap>'    CC:             gcc'</span></li><li><span style=white-space:pre-wrap>'    CPPFLAGS:       -frelease  -Wdate-time -D_FORTIFY_SOURCE=2'</span></li><li><span style=white-space:pre-wrap>'    CFLAGS:         -g -O2 -ffile-prefix-map=\\xabBUILDPATH\\xbb=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security'</span></li><li><span style=white-space:pre-wrap>'    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now'</span></li><li><span style=white-space:pre-wrap>'    HTSDIR:         '</span></li><li><span style=white-space:pre-wrap>'    LIBS:           '</span></li><li><span style=white-space:pre-wrap>'    CURSES_LIB:     -lcurses'</span></li><li>''</li><li>'HTSlib compilation details:'</li><li><span style=white-space:pre-wrap>'    Features:       build=configure plugins=yes, plugin-path=/usr/local/lib/htslib:/usr/local/libexec/htslib:: libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes htscodecs=1.1.1'</span></li><li><span style=white-space:pre-wrap>'    CC:             gcc'</span></li><li><span style=white-space:pre-wrap>'    CPPFLAGS:       -I. -DSAMTOOLS=1 -Wdate-time -D_FORTIFY_SOURCE=2'</span></li><li><span style=white-space:pre-wrap>'    CFLAGS:         -g -O2 -ffile-prefix-map=/build/htslib-TQtOKr/htslib-1.13+ds=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -flto -fvisibility=hidden -flto -fvisibility=hidden'</span></li><li><span style=white-space:pre-wrap>'    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,-flto -fvisibility=hidden -Wl,-flto -fvisibility=hidden'</span></li><li>''</li><li>'HTSlib URL scheme handlers present:'</li><li><span style=white-space:pre-wrap>'    built-in:\\t preload, data, file'</span></li><li><span style=white-space:pre-wrap>'    crypt4gh-needed:\\t crypt4gh'</span></li><li><span style=white-space:pre-wrap>'    mem:\\t mem'</span></li></ol>\n"
            ],
            "text/markdown": "1. 'samtools 1.13'\n2. 'Using htslib 1.13+ds'\n3. 'Copyright (C) 2021 Genome Research Ltd.'\n4. ''\n5. 'Samtools compilation details:'\n6. <span style=white-space:pre-wrap>'    Features:       build=configure curses=yes '</span>\n7. <span style=white-space:pre-wrap>'    CC:             gcc'</span>\n8. <span style=white-space:pre-wrap>'    CPPFLAGS:       -frelease  -Wdate-time -D_FORTIFY_SOURCE=2'</span>\n9. <span style=white-space:pre-wrap>'    CFLAGS:         -g -O2 -ffile-prefix-map=\\xabBUILDPATH\\xbb=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security'</span>\n10. <span style=white-space:pre-wrap>'    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now'</span>\n11. <span style=white-space:pre-wrap>'    HTSDIR:         '</span>\n12. <span style=white-space:pre-wrap>'    LIBS:           '</span>\n13. <span style=white-space:pre-wrap>'    CURSES_LIB:     -lcurses'</span>\n14. ''\n15. 'HTSlib compilation details:'\n16. <span style=white-space:pre-wrap>'    Features:       build=configure plugins=yes, plugin-path=/usr/local/lib/htslib:/usr/local/libexec/htslib:: libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes htscodecs=1.1.1'</span>\n17. <span style=white-space:pre-wrap>'    CC:             gcc'</span>\n18. <span style=white-space:pre-wrap>'    CPPFLAGS:       -I. -DSAMTOOLS=1 -Wdate-time -D_FORTIFY_SOURCE=2'</span>\n19. <span style=white-space:pre-wrap>'    CFLAGS:         -g -O2 -ffile-prefix-map=/build/htslib-TQtOKr/htslib-1.13+ds=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -flto -fvisibility=hidden -flto -fvisibility=hidden'</span>\n20. <span style=white-space:pre-wrap>'    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,-flto -fvisibility=hidden -Wl,-flto -fvisibility=hidden'</span>\n21. ''\n22. 'HTSlib URL scheme handlers present:'\n23. <span style=white-space:pre-wrap>'    built-in:\\t preload, data, file'</span>\n24. <span style=white-space:pre-wrap>'    crypt4gh-needed:\\t crypt4gh'</span>\n25. <span style=white-space:pre-wrap>'    mem:\\t mem'</span>\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item 'samtools 1.13'\n\\item 'Using htslib 1.13+ds'\n\\item 'Copyright (C) 2021 Genome Research Ltd.'\n\\item ''\n\\item 'Samtools compilation details:'\n\\item '    Features:       build=configure curses=yes '\n\\item '    CC:             gcc'\n\\item '    CPPFLAGS:       -frelease  -Wdate-time -D\\_FORTIFY\\_SOURCE=2'\n\\item '    CFLAGS:         -g -O2 -ffile-prefix-map=\\textbackslash{}xabBUILDPATH\\textbackslash{}xbb=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security'\n\\item '    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now'\n\\item '    HTSDIR:         '\n\\item '    LIBS:           '\n\\item '    CURSES\\_LIB:     -lcurses'\n\\item ''\n\\item 'HTSlib compilation details:'\n\\item '    Features:       build=configure plugins=yes, plugin-path=/usr/local/lib/htslib:/usr/local/libexec/htslib:: libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes htscodecs=1.1.1'\n\\item '    CC:             gcc'\n\\item '    CPPFLAGS:       -I. -DSAMTOOLS=1 -Wdate-time -D\\_FORTIFY\\_SOURCE=2'\n\\item '    CFLAGS:         -g -O2 -ffile-prefix-map=/build/htslib-TQtOKr/htslib-1.13+ds=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -flto -fvisibility=hidden -flto -fvisibility=hidden'\n\\item '    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,-flto -fvisibility=hidden -Wl,-flto -fvisibility=hidden'\n\\item ''\n\\item 'HTSlib URL scheme handlers present:'\n\\item '    built-in:\\textbackslash{}t preload, data, file'\n\\item '    crypt4gh-needed:\\textbackslash{}t crypt4gh'\n\\item '    mem:\\textbackslash{}t mem'\n\\end{enumerate*}\n",
            "text/plain": [
              " [1] \"samtools 1.13\"                                                                                                                                                                                                                \n",
              " [2] \"Using htslib 1.13+ds\"                                                                                                                                                                                                         \n",
              " [3] \"Copyright (C) 2021 Genome Research Ltd.\"                                                                                                                                                                                      \n",
              " [4] \"\"                                                                                                                                                                                                                             \n",
              " [5] \"Samtools compilation details:\"                                                                                                                                                                                                \n",
              " [6] \"    Features:       build=configure curses=yes \"                                                                                                                                                                              \n",
              " [7] \"    CC:             gcc\"                                                                                                                                                                                                      \n",
              " [8] \"    CPPFLAGS:       -frelease  -Wdate-time -D_FORTIFY_SOURCE=2\"                                                                                                                                                               \n",
              " [9] \"    CFLAGS:         -g -O2 -ffile-prefix-map=\\xabBUILDPATH\\xbb=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security\"                                                                      \n",
              "[10] \"    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now\"                                                                                                                                              \n",
              "[11] \"    HTSDIR:         \"                                                                                                                                                                                                         \n",
              "[12] \"    LIBS:           \"                                                                                                                                                                                                         \n",
              "[13] \"    CURSES_LIB:     -lcurses\"                                                                                                                                                                                                 \n",
              "[14] \"\"                                                                                                                                                                                                                             \n",
              "[15] \"HTSlib compilation details:\"                                                                                                                                                                                                  \n",
              "[16] \"    Features:       build=configure plugins=yes, plugin-path=/usr/local/lib/htslib:/usr/local/libexec/htslib:: libcurl=yes S3=yes GCS=yes libdeflate=yes lzma=yes bzip2=yes htscodecs=1.1.1\"                                  \n",
              "[17] \"    CC:             gcc\"                                                                                                                                                                                                      \n",
              "[18] \"    CPPFLAGS:       -I. -DSAMTOOLS=1 -Wdate-time -D_FORTIFY_SOURCE=2\"                                                                                                                                                         \n",
              "[19] \"    CFLAGS:         -g -O2 -ffile-prefix-map=/build/htslib-TQtOKr/htslib-1.13+ds=. -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -flto -fvisibility=hidden -flto -fvisibility=hidden\"\n",
              "[20] \"    LDFLAGS:        -Wl,-Bsymbolic-functions -flto=auto -Wl,-z,relro -Wl,-z,now -Wl,-flto -fvisibility=hidden -Wl,-flto -fvisibility=hidden\"                                                                                  \n",
              "[21] \"\"                                                                                                                                                                                                                             \n",
              "[22] \"HTSlib URL scheme handlers present:\"                                                                                                                                                                                          \n",
              "[23] \"    built-in:\\t preload, data, file\"                                                                                                                                                                                          \n",
              "[24] \"    crypt4gh-needed:\\t crypt4gh\"                                                                                                                                                                                              \n",
              "[25] \"    mem:\\t mem\"                                                                                                                                                                                                               "
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Indexar tu BAM inmediatamente\n",
        "system(\"samtools index /content/mito_fixed.bam\")"
      ],
      "metadata": {
        "id": "rlJuucnxxwsJ"
      },
      "execution_count": 16,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Instalar gdown (ignora archivos grandes de Drive perfectamente)\n",
        "system(\"pip install gdown\")\n",
        "\n",
        "# 2. Descargar tu BAM de 27MB CORRECTAMENTE\n",
        "system(\"gdown 19hqh2T5YGlpIzOwn7Nkcw6CwOfn8KoeN -O /content/mito_fixed.bam --fuzzy\")\n",
        "\n",
        "# 3. Verificar descarga REAL\n",
        "system(\"ls -lh /content/mito*.bam\")\n",
        "system(\"samtools quickcheck /content/mito_fixed.bam\")\n"
      ],
      "metadata": {
        "id": "ZGA1qYP8wZdV"
      },
      "execution_count": 15,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Listar TODOS los archivos mito en /content\n",
        "system(\"ls -lh /content/mito.*\", intern = TRUE)\n",
        "\n",
        "# 2. Si NO hay mito.bam.bai → diagnosticar BAM\n",
        "system(\"samtools quickcheck /content/mito_fixed.bam\", intern = TRUE)\n",
        "system(\"samtools view -H /content/mito_fixed.bam | head -5\", intern = TRUE)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 124
        },
        "id": "fCS08z4gfBJ6",
        "outputId": "06ff4ea0-15c0-49d0-f40c-9e7d8af28de5"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Warning message in system(\"ls -lh /content/mito.*\", intern = TRUE):\n",
            "“running command 'ls -lh /content/mito.*' had status 2”\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)\n",
              "attr(,\"status\")\n",
              "[1] 2\n",
              "attr(,\"errmsg\")\n",
              "[1] \"Resource temporarily unavailable\""
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [],
            "text/markdown": "",
            "text/latex": "",
            "text/plain": [
              "character(0)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'@HD\\tVN:1.6\\tSO:coordinate'</li><li>'@SQ\\tSN:1\\tLN:249250621'</li><li>'@SQ\\tSN:2\\tLN:243199373'</li><li>'@SQ\\tSN:3\\tLN:198022430'</li><li>'@SQ\\tSN:4\\tLN:191154276'</li></ol>\n"
            ],
            "text/markdown": "1. '@HD\\tVN:1.6\\tSO:coordinate'\n2. '@SQ\\tSN:1\\tLN:249250621'\n3. '@SQ\\tSN:2\\tLN:243199373'\n4. '@SQ\\tSN:3\\tLN:198022430'\n5. '@SQ\\tSN:4\\tLN:191154276'\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item '@HD\\textbackslash{}tVN:1.6\\textbackslash{}tSO:coordinate'\n\\item '@SQ\\textbackslash{}tSN:1\\textbackslash{}tLN:249250621'\n\\item '@SQ\\textbackslash{}tSN:2\\textbackslash{}tLN:243199373'\n\\item '@SQ\\textbackslash{}tSN:3\\textbackslash{}tLN:198022430'\n\\item '@SQ\\textbackslash{}tSN:4\\textbackslash{}tLN:191154276'\n\\end{enumerate*}\n",
            "text/plain": [
              "[1] \"@HD\\tVN:1.6\\tSO:coordinate\" \"@SQ\\tSN:1\\tLN:249250621\"   \n",
              "[3] \"@SQ\\tSN:2\\tLN:243199373\"    \"@SQ\\tSN:3\\tLN:198022430\"   \n",
              "[5] \"@SQ\\tSN:4\\tLN:191154276\"   "
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# MODO VERBOSE TOTAL - NO capturar NADA\n",
        "system2(\"samtools\", args=c(\"flagstat\", \"/content/mito_fixed.bam\"), stdout=TRUE, stderr=TRUE)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 191
        },
        "id": "ekQqRKy850eZ",
        "outputId": "84486ca0-a335-4990-d27d-934d495dec04"
      },
      "execution_count": 18,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<style>\n",
              ".list-inline {list-style: none; margin:0; padding: 0}\n",
              ".list-inline>li {display: inline-block}\n",
              ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
              "</style>\n",
              "<ol class=list-inline><li>'396988 + 0 in total (QC-passed reads + QC-failed reads)'</li><li>'396507 + 0 primary'</li><li>'0 + 0 secondary'</li><li>'481 + 0 supplementary'</li><li>'26702 + 0 duplicates'</li><li>'26702 + 0 primary duplicates'</li><li>'396155 + 0 mapped (99.79% : N/A)'</li><li>'395674 + 0 primary mapped (99.79% : N/A)'</li><li>'396507 + 0 paired in sequencing'</li><li>'198307 + 0 read1'</li><li>'198200 + 0 read2'</li><li>'386506 + 0 properly paired (97.48% : N/A)'</li><li>'394841 + 0 with itself and mate mapped'</li><li>'833 + 0 singletons (0.21% : N/A)'</li><li>'4603 + 0 with mate mapped to a different chr'</li><li>'4322 + 0 with mate mapped to a different chr (mapQ&gt;=5)'</li></ol>\n"
            ],
            "text/markdown": "1. '396988 + 0 in total (QC-passed reads + QC-failed reads)'\n2. '396507 + 0 primary'\n3. '0 + 0 secondary'\n4. '481 + 0 supplementary'\n5. '26702 + 0 duplicates'\n6. '26702 + 0 primary duplicates'\n7. '396155 + 0 mapped (99.79% : N/A)'\n8. '395674 + 0 primary mapped (99.79% : N/A)'\n9. '396507 + 0 paired in sequencing'\n10. '198307 + 0 read1'\n11. '198200 + 0 read2'\n12. '386506 + 0 properly paired (97.48% : N/A)'\n13. '394841 + 0 with itself and mate mapped'\n14. '833 + 0 singletons (0.21% : N/A)'\n15. '4603 + 0 with mate mapped to a different chr'\n16. '4322 + 0 with mate mapped to a different chr (mapQ&gt;=5)'\n\n\n",
            "text/latex": "\\begin{enumerate*}\n\\item '396988 + 0 in total (QC-passed reads + QC-failed reads)'\n\\item '396507 + 0 primary'\n\\item '0 + 0 secondary'\n\\item '481 + 0 supplementary'\n\\item '26702 + 0 duplicates'\n\\item '26702 + 0 primary duplicates'\n\\item '396155 + 0 mapped (99.79\\% : N/A)'\n\\item '395674 + 0 primary mapped (99.79\\% : N/A)'\n\\item '396507 + 0 paired in sequencing'\n\\item '198307 + 0 read1'\n\\item '198200 + 0 read2'\n\\item '386506 + 0 properly paired (97.48\\% : N/A)'\n\\item '394841 + 0 with itself and mate mapped'\n\\item '833 + 0 singletons (0.21\\% : N/A)'\n\\item '4603 + 0 with mate mapped to a different chr'\n\\item '4322 + 0 with mate mapped to a different chr (mapQ>=5)'\n\\end{enumerate*}\n",
            "text/plain": [
              " [1] \"396988 + 0 in total (QC-passed reads + QC-failed reads)\"\n",
              " [2] \"396507 + 0 primary\"                                     \n",
              " [3] \"0 + 0 secondary\"                                        \n",
              " [4] \"481 + 0 supplementary\"                                  \n",
              " [5] \"26702 + 0 duplicates\"                                   \n",
              " [6] \"26702 + 0 primary duplicates\"                           \n",
              " [7] \"396155 + 0 mapped (99.79% : N/A)\"                       \n",
              " [8] \"395674 + 0 primary mapped (99.79% : N/A)\"               \n",
              " [9] \"396507 + 0 paired in sequencing\"                        \n",
              "[10] \"198307 + 0 read1\"                                       \n",
              "[11] \"198200 + 0 read2\"                                       \n",
              "[12] \"386506 + 0 properly paired (97.48% : N/A)\"              \n",
              "[13] \"394841 + 0 with itself and mate mapped\"                 \n",
              "[14] \"833 + 0 singletons (0.21% : N/A)\"                       \n",
              "[15] \"4603 + 0 with mate mapped to a different chr\"           \n",
              "[16] \"4322 + 0 with mate mapped to a different chr (mapQ>=5)\" "
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "\n",
        "# 1. VER ERROR COMPLETO de HaploCart (¡clave!)\n",
        "system(paste0(ruta_vgan, \" haplocart -i /content/mito_fixed.bam -o haplogrupo_debug.tsv --hcfiles ~/share/vgan/hcfiles/ 2>&1\"), intern = FALSE)\n",
        "\n",
        "# 2. Verificar hcfiles descargados\n",
        "system(\"ls -lh ~/share/vgan/hcfiles/\", intern = TRUE)\n",
        "\n",
        "# 3. Contenido del BAM (¿hay lecturas mtDNA?)\n",
        "system(\"samtools view /content/mito_fixed.bam | head -5\", intern = FALSE)\n",
        "\n",
        "# 4. Verificar índice\n",
        "system(\"ls -lh /content/mito_fixed.bam*\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 34
        },
        "id": "r-8GjD0dyrxu",
        "outputId": "ed82329f-08cf-4e7c-a03f-3d7bba48cefc"
      },
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "'total 0'"
            ],
            "text/markdown": "'total 0'",
            "text/latex": "'total 0'",
            "text/plain": [
              "[1] \"total 0\""
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Check estado runtime\n",
        "system(\"nproc\")  # Si sale \"8\" → runtime OK\n",
        "system(\"df -h /content\")  # Espacio disco\n"
      ],
      "metadata": {
        "id": "13QPHPpy3853"
      },
      "execution_count": 20,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# CONTENIDO REAL del BAM - ¡esto lo explica TODO!\n",
        "system(\"samtools flagstat /content/mito_fixed.bam\")\n",
        "system(\"samtools view /content/mito_fixed.bam | wc -l\")  # Número lecturas\n",
        "system(\"samtools view /content/mito_fixed.bam | head -3\") # Primeras lecturas\n"
      ],
      "metadata": {
        "id": "R-OaSvwM4dEj"
      },
      "execution_count": 21,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# VER LECTURAS REALES (SIN SILENCIO)\n",
        "system(\"samtools view /content/mito_fixed.bam | head -3\", intern = FALSE)\n",
        "system(\"samtools view /content/mito_fixed.bam | wc -l\", intern = FALSE)\n"
      ],
      "metadata": {
        "id": "Xv7pXG1I4sqV"
      },
      "execution_count": 22,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# MOSTRAR TODO en consola (NO capturar)\n",
        "system(\"samtools flagstat /content/mito_fixed.bam\", intern = FALSE)\n",
        "system(\"samtools view /content/mito_fixed.bam | head -3\", intern = FALSE)\n"
      ],
      "metadata": {
        "id": "q7uxLcIX5Dmm"
      },
      "execution_count": 7,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "\n",
        "# HAPLOCART con TODOS los errores a pantalla\n",
        "system(paste0(ruta_vgan, \" haplocart -i /content/mito_fixed.bam -o debug_verbose.tsv --hcfiles ~/share/vgan/hcfiles/ 2>&1 | tee haplocart_debug.log\"), intern = FALSE)\n"
      ],
      "metadata": {
        "id": "KLInjKnh6deL"
      },
      "execution_count": 10,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "\n",
        "# 1. Crear hcfiles en la ubicación EXACTA que busca HaploCart\n",
        "system(\"mkdir -p /tmp/vgan-env/envs/vgan/share/hcfiles\")\n",
        "\n",
        "# 2. Descargar hcfiles en esa ubicación específica\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles && wget https://github.com/grenaud/vgan/raw/main/hcfiles/human_mtdna_hg.pb\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles && wget https://github.com/grenaud/vgan/raw/main/hcfiles/human_mtdna_hg.gcsa\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles && wget https://github.com/grenaud/vgan/raw/main/hcfiles/human_mtdna_hg.gcsa.lcp\")\n",
        "\n",
        "# 3. ¡EJECUTAR HAPLOCART FINAL!\n",
        "system(paste0(ruta_vgan, \" haplocart -i /content/mito_fixed.bam -o haplogrupo_CORRECTO.tsv\"), intern = FALSE)\n",
        "\n",
        "# 4. VER TU HAPLOGUPO\n",
        "system(\"cat haplogrupo_CORRECTO.tsv\")\n"
      ],
      "metadata": {
        "id": "V1dvPGmU690F"
      },
      "execution_count": 11,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Fix FINAL (solo faltan hcfiles en ruta correcta)\n",
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "system(\"mkdir -p /tmp/vgan-env/envs/vgan/share/hcfiles\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles && wget https://github.com/grenaud/vgan/raw/main/hcfiles/human_mtdna_hg.pb\")\n",
        "system(\"ls -lh /tmp/vgan-env/envs/vgan/share/hcfiles/\")\n",
        "system(paste0(ruta_vgan, \" haplocart -i /content/mito_fixed.bam -o FINAL.tsv\"))\n",
        "system(\"cat FINAL.tsv\")  # ¡TU HAPLOGUPO!\n"
      ],
      "metadata": {
        "id": "SSCclalp7Zgy"
      },
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "system(\"ls -lh /tmp/vgan-env/envs/vgan/share/hcfiles/\", intern = FALSE)\n",
        "system(paste0(ruta_vgan, \" haplocart --help | head -20\"), intern = FALSE)\n"
      ],
      "metadata": {
        "id": "DTJxi3eR_ZeQ"
      },
      "execution_count": 23,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# ÚNICO COMANDO que funciona (sin micromamba roto)\n",
        "system(\"docker run -v /content:/data grenaud/vgan:latest haplocart -i /content/mito_fixed.bam -o /content/haplogrupo.tsv\")\n",
        "system(\"cat /content/haplogrupo.tsv\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-rhKkPOKUUN4",
        "outputId": "7935a7a4-1237-40bc-bd5b-a0c4a3b4a921"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Warning message in system(\"docker run -v /content:/data grenaud/vgan:latest haplocart -i /content/mito_fixed.bam -o /content/haplogrupo.tsv\"):\n",
            "“error in running command”\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. Verificar BAM OK\n",
        "system(\"samtools flagstat /content/mito_fixed.bam\", intern = FALSE)\n",
        "\n",
        "# 2. Crear directorio EXACTO que HaploCart espera\n",
        "system(\"mkdir -p /tmp/vgan-env/envs/vgan/share/hcfiles\")\n",
        "\n",
        "# 3. Descargar hcfiles desde FUENTE OFICIAL HaploCart\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles && curl -L -o human_mtdna_hg.pb 'https://services.healthtech.dtu.dk/services/HaploCart-1.0/human_mtdna_hg.pb'\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles && curl -L -o human_mtdna_hg.gcsa 'https://services.healthtech.dtu.dk/services/HaploCart-1.0/human_mtdna_hg.gcsa'\")\n",
        "\n",
        "# 4. Verificar hcfiles descargados\n",
        "system(\"ls -lh /tmp/vgan-env/envs/vgan/share/hcfiles/\", intern = FALSE)\n",
        "\n",
        "# 5. EJECUTAR SIN --hcfiles (usa ruta interna)\n",
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "system(paste(ruta_vgan, \"haplocart -i /content/mito_fixed.bam -o /content/MI_HAPLOGUPO.tsv 2>&1 | tee haplocart.log\"), intern = FALSE)\n",
        "\n",
        "# 6. RESULTADO FINAL\n",
        "system(\"cat /content/MI_HAPLOGUPO.tsv\")\n",
        "system(\"cat haplocart.log | tail -10\")\n"
      ],
      "metadata": {
        "id": "xbRY_99eUhWM"
      },
      "execution_count": 25,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. LIMPIAR archivos corruptos\n",
        "system(\"rm -rf ~/bin/vgan ~/share/vgan\")\n",
        "\n",
        "# 2. hcfiles OFICIALES en ruta EXACTA\n",
        "system(\"mkdir -p /tmp/vgan-env/envs/vgan/share/hcfiles\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles/ && curl -L -o human_mtdna_hg.pb 'https://services.healthtech.dtu.dk/services/HaploCart-1.0/human_mtdna_hg.pb'\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles/ && curl -L -o human_mtdna_hg.gcsa 'https://services.healthtech.dtu.dk/services/HaploCart-1.0/human_mtdna_hg.gcsa'\")\n",
        "system(\"ls -lh /tmp/vgan-env/envs/vgan/share/hcfiles/\", intern=FALSE)\n",
        "\n",
        "# 3. ¡EJECUTAR HAPLOCART!\n",
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "system(paste(ruta_vgan, \"haplocart -i /content/mito_fixed.bam -o /content/MI_HAPLOGUPO.tsv 2>&1 | tee haplocart.log\"), intern=FALSE)\n",
        "\n",
        "# 4. TU HAPLOGUPO\n",
        "system(\"cat /content/MI_HAPLOGUPO.tsv\")\n"
      ],
      "metadata": {
        "id": "TDWG2fw5VZ4R"
      },
      "execution_count": 26,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# 1. hcfiles OFICIALES en ruta EXACTA\n",
        "system(\"mkdir -p /tmp/vgan-env/envs/vgan/share/hcfiles\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles/ && curl -L -o human_mtdna_hg.pb 'https://services.healthtech.dtu.dk/services/HaploCart-1.0/human_mtdna_hg.pb'\")\n",
        "system(\"cd /tmp/vgan-env/envs/vgan/share/hcfiles/ && curl -L -o human_mtdna_hg.gcsa 'https://services.healthtech.dtu.dk/services/HaploCart-1.0/human_mtdna_gcsa'\")\n",
        "\n",
        "# 2. ¡TU HAPLOGUPO!\n",
        "ruta_vgan <- \"/tmp/vgan-env/envs/vgan/bin/vgan\"\n",
        "system(paste(ruta_vgan, \"haplocart -i /content/mito_fixed.bam -o /content/MI_HAPLOGUPO.tsv\"), intern=FALSE)\n",
        "system(\"cat /content/MI_HAPLOGUPO.tsv\")\n"
      ],
      "metadata": {
        "id": "xyisiOKdVvLH"
      },
      "execution_count": 27,
      "outputs": []
    }
  ]
}